nixos/modules/services/web-servers/garage.nix

```diff
@@ -36,6 +36,12 @@ in
       description = "Garage log level, see <https://garagehq.deuxfleurs.fr/documentation/quick-start/#launching-the-garage-server> for examples.";
     };
 
+    systemdLoadCredential = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = "Optional load credential for systemd. Useful if you want to use sops or agenix to manage your secrets.";
+    };
+
```

Comment:
If this just passes these values directly to systemd, I don't see too much value with this.

Reply:
It's not just passing the value. The point is to be able to pass a file. 
Here is an example that I'm using : 
```    
systemdLoadCredential = [
      "admin:${config.age.secrets.garage_admin_token.path}"
      "metrics:${config.age.secrets.garage_metrics_token.path}"
      "rpc:${config.age.secrets.garage_rpc_secret.path}"
];
```
It allows you to pass the credential and the file containing the secrets. I did not find better solution to use agenix for the secrets.

Reply:
Sorry, my point was that users could just use `systemd.services.garage.serviceConfig.LoadCredential = []` directly. I think this would be more valuable if there was an option to load the credential and set the env var at the same time.
For example:

```nix
{
  services.garage.credentialFiles = {
    GARAGE_RPC_SECRET_FILE = config.age.secrets.garage_rpc_secret.path;
    GARAGE_ADMIN_TOKEN_FILE = config.age.secrets.garage_admin_token.path;
    GARAGE_METRICS_TOKEN_FILE = config.age.secrets.garage_metrics_token.path;
  };
}
```

Perhaps the name of this option should be more descriptive, though. This was just an idea ^^